src/converter/one/convert/JfrToOtlp.java

```diff
@@ -0,0 +1,275 @@
+
+    private final Proto proto = new Proto(1024);
+
+    public JfrToOtlp(JfrReader jfr, Arguments args) {
+        super(jfr, args);
+
+        // string_table[0] must always be the empty string
+        stringPool.index("");
+    }
+
+    @Override
+    public void convert() throws IOException {
+        long rpMark = proto.startField(PROFILES_DATA_resource_profiles, BIG_MESSAGE_BYTE_COUNT);
+        long spMark = proto.startField(RESOURCE_PROFILES_scope_profiles, BIG_MESSAGE_BYTE_COUNT);
+        super.convert();
+        proto.commitField(spMark);
+        proto.commitField(rpMark);
+
+        writeProfileDictionary();
+    }
+
+    @Override
+    protected void convertChunk() {
+        long pMark = proto.startField(SCOPE_PROFILES_profiles, BIG_MESSAGE_BYTE_COUNT);
+
+        writeSampleTypes();
+        writeTimingInformation();
+
+        List<Integer> locationIndices = new ArrayList<>();
+        collector.forEach(new OtlpEventToSampleVisitor(locationIndices));
+
+        long locationIndicesMark =
+                proto.startField(PROFILE_location_indices, BIG_MESSAGE_BYTE_COUNT);
+        locationIndices.forEach(proto::writeInt);
+        proto.commitField(locationIndicesMark);
+
+        proto.commitField(pMark);
+    }
+
+    private void writeSampleTypes() {
+        long stsMark = proto.startField(PROFILE_sample_type, 1);
+        proto.field(VALUE_TYPE_type_strindex, stringPool.index(getValueType()));
+        proto.field(VALUE_TYPE_unit_strindex, stringPool.index(getSampleUnits()));
+        proto.field(VALUE_TYPE_aggregation_temporality, AGGREGATION_TEMPORARALITY_cumulative);
+        proto.commitField(stsMark);
+
+        long sttMark = proto.startField(PROFILE_sample_type, 1);
+        proto.field(VALUE_TYPE_type_strindex, stringPool.index(getValueType()));
+        proto.field(VALUE_TYPE_unit_strindex, stringPool.index(getTotalUnits()));
+        proto.field(VALUE_TYPE_aggregation_temporality, AGGREGATION_TEMPORARALITY_cumulative);
+        proto.commitField(sttMark);
+    }
+
+    private void writeTimingInformation() {
+        proto.field(PROFILE_time_nanos, jfr.getChunkStartNanos());
+        proto.field(PROFILE_duration_nanos, jfr.chunkDurationNanos());
+    }
+
+    public void dump(OutputStream out) throws IOException {
+        out.write(proto.buffer(), 0, proto.size());
+    }
+
+    private void writeProfileDictionary() {
+        long profilesDictionaryMark =
+                proto.startField(PROFILES_DATA_dictionary, BIG_MESSAGE_BYTE_COUNT);
+
+        // Mapping[0] must be a default mapping according to the spec
+        long mappingMark = proto.startField(PROFILES_DICTIONARY_mapping_table, 1);
+        proto.commitField(mappingMark);
+
+        // Write function table
+        for (String name : functionPool.keys()) {
+            long fMark = proto.startField(PROFILES_DICTIONARY_function_table, 1);
+            int nameStrindex = stringPool.index(name);
```

Comment:
do you really need `nameStrindex` variable?
```suggestion
            proto.field(FUNCTION_name_strindex, stringPool.index(name));
```